transports/http/endpoints/api/merkleroots/endpoints.go

```diff
@@ -25,6 +33,44 @@ func (h *handler) RegisterAPIEndpoints(router *gin.RouterGroup, _ *config.HTTPCo
 	merkle := router.Group("/chain/merkleroot")
 	{
 		merkle.POST("/verify", h.verify)
+		merkle.GET("", h.merkleroots)
+	}
+}
+
+// Merkleroot godoc.
+//
+// @Summary Returns merkleroots for the specified range
+// @Tags merkleroots
+// @Accept */*
+// @Produce json
+// @Success 200 {object} merkleroots.MerkleRootsESKPagedResponse.//TODO: map this in swagger docs
+// @Router /chain/merkleroot [get]
+// @Param batchSize query string false "Batch size of returned merkleroots"
+// @Param lastEvaluatedKey query string false "Last evaluated block height that client has processed"
+// @Security Bearer
+func (h *handler) merkleroots(c *gin.Context) {
+	batchSize := c.DefaultQuery("batchSize", batchSize)
+	lastEvaluatedKey := c.DefaultQuery("lastEvaluatedKey", lastEvaluatedKey)
+
+	batchSizeInt, errBatchSizeConv := strconv.Atoi(batchSize)
+	lastEvaluatedKeyInt, errLastEvaluatedKeyConv := strconv.Atoi(lastEvaluatedKey)
+
+	if errBatchSizeConv != nil {
+		c.JSON(http.StatusBadRequest, errors.New("batchSize must be a numeric value").Error())
+		return
+	}
+
+	if errLastEvaluatedKeyConv != nil {
+		c.JSON(http.StatusBadRequest, errors.New("lastEvaluatedKey must be a numeric value").Error())
+		return
+	}
```

Comment:
Same as my one of the previous comment.
Try to return err as soon as possible. And try to avoid custom names of errors (if possible) - to not confuse other developers.

```suggestion
	batchSizeInt, err := strconv.Atoi(batchSize)
	if err != nil {
		c.JSON(http.StatusBadRequest, errors.New("batchSize must be a numeric value").Error())
		return
	}

	lastEvaluatedKeyInt, err := strconv.Atoi(lastEvaluatedKey)
	if err != nil {
		c.JSON(http.StatusBadRequest, errors.New("lastEvaluatedKey must be a numeric value").Error())
		return
	}
```